Profile page

alshakerM opened the pull request:
Added profile page

alshakero commented on src/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,275 @@
+                <button className={styles.threeDotsButton}>
+                  <svg
+                    aria-label="Options"
+                    class="_8-yf5 "
```
Unused class.

alshakero reviewed (changes requested):
1. It seems you're not mapping the user posts and always assuming that the user has one post (by using `posts[0]`). Currently, we have one post per user because the data is flawed, but users can have 0 to 1000 posts. You should do `posts.map(...)`. And should handle the case when users have zero posts as well.

2. It seems you're using different HTML for posts and tagged posts, even though the post items in both "Posts" and "Tagged" section look exactly the same. You need a component to make the code DRY (maybe `UserProfilePost`).

alshakerM commented:
Created an icon component.
Changed to the code in normal posts to map.
Added a section if there were no posts available for post and tagged post.
Removed an unneeded class from an icon.

alshakero commented on src/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,237 @@
+        <div className={styles.header}>
+          <div className={styles.imgContainer}>
+            <Link to={`/stories/${user?.user_id}`}>
+              <img
+                src={user?.user_thumbnail}
+                alt={`${user?.user_name} avatar`}
+                className={styles.userImg}
+              />
```
Why not reuse Avatar?

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  color: #8e8e8e;
+  height: 52px;
+}
+.isClicked {
+  color: #262626;
+  border-top: 1px solid #262626;
+}
```
Should be `isSelected`. It can be opened without clicking.

alshakerM replied:
Oh yea, great catch.
On it

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+.postsSectionPost {
+  display: flex;
+}
+.postsSectionPost:hover .postImg {
+  position: static;
+}
```
I'm not sure, but I'm not sure you need this.

alshakerM replied:
You were right, I don't.

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+.postsSectionPost:hover .postImg {
+  position: static;
+}
+.taggedPost:hover .isTagged {
+  position: relative;
+}
```
This can go.

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  font-weight: 600;
+  visibility: hidden;
+}
+.isTagged {
+  margin: 0;
+  margin-top: -296px;
+}
```
```suggestion
.isTagged {
  margin: 0;
  margin-top: -296px;
}
```

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  flex-wrap: wrap;
+  gap: 28px;
+}
+.emptyPostTextSection {
+  width: 555px;
+  height: 380px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  background-color: #fff;
+  align-items: center;
+}
```
This is unused.

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  background-color: #fff;
+  align-items: center;
+}
+.emptyPostText {
+  font-size: 18px;
+  margin: 0;
+}
```
+1

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  font-size: 18px;
+  margin: 0;
+}
+.emptyPostAppText {
+  font-size: 16px;
+  margin: 0;
+  margin-top: 10px;
+  margin-bottom: 25px;
+}
```
Unused.

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  margin-top: 10px;
+  margin-bottom: 25px;
+}
+.storeImagesSection {
+  display: flex;
+  gap: 8px;
+}
```
Unused

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,235 @@
+  display: flex;
+  gap: 8px;
+}
+.appleStoreImg,
+.googleStoreImg {
+  width: 136px;
+  height: 40px;
+}
+.randomImg {
+  width: 380px;
+  height: 380px;
+}
```
+1

alshakero reviewed (changes requested):
Looks fabulous. Left some important comments.

alshakerM commented:
Ready for another round.

alshakero commented on src/App.js:
```diff
@@ -27,6 +28,13 @@ function App() {
         path="/stories/:userId/"
         render={(match) => <UserStories userId={match.match.params.userId} />}
       ></Route>
+      <Route
+        path={['/:userName/', '/:userName/channel/', '/:userName/tagged/']}
+        exact
+        render={(match) => (
+          <UserProfile userName={match.match.params.userName} />
+        )}
+      ></Route>
```
Since this route accepts `/anything` (see the first path) you have to put it last in the Switch. Let me know if you don't understand this.

alshakero commented on src/UserProfile/UserProfile.module.css:
```diff
@@ -0,0 +1,195 @@
+  align-items: center;
+  gap: 7px;
+}
+.Icon {
+  fill: #fff;
+}
```
```suggestion
.icon {
  fill: #fff;
}
```

alshakero reviewed (approved):
Perfect. Left some cosmectic comments. Feel free to merge after addressing.
